src/amo/withExperiment.js

```diff
@@ -167,49 +173,52 @@ export const withExperiment = ({
     constructor(props: withExperimentInternalProps) {
       super(props);
 
-      const {
-        _getVariant,
-        _isExperimentEnabled,
-        cookies,
-        id,
-        variants,
-      } = this.props;
-
-      const isEnabled = _isExperimentEnabled({ _config, id });
-      const registeredExperiments = this.getExperiments();
-      const experimentInCookie = this.cookieIncludesExperiment(
-        registeredExperiments,
-      );
-      const experimentsToStore = { ...registeredExperiments };
-
-      // Clear any disabled experiments from the cookie.
-      let cleanupNeeded = false;
-      for (const experimentId of Object.keys(registeredExperiments)) {
-        if (!_isExperimentEnabled({ _config, id: experimentId })) {
-          delete experimentsToStore[experimentId];
-          cleanupNeeded = true;
+      // We only want to run this code if we are on the client.
+      if (!_config.get('server')) {
```

Comment:
Can we use `componentDidMount()` instead of the `constructor()` or is it too late?

Reply:
We can, and it does seem to work, but I'm not sure how to write a test for the case where we're not running on the client. I guess we don't do that for any of our components that use `componentDidMount`, and just rely on the fact that it doesn't run on the server. 

This all seems a bit fragile to me, and non-explicit, but I guess it's the way we've always done it, and we rely on that behaviour in a lot of places. I wonder what we're going to do if we ever move to hooks? I guess that's an issue for the future AMO team.

I can change it and remove the tests that verify it doesn't run on the server.

Reply:
do you think we could write an integration test like we do here? https://github.com/mozilla/addons-frontend/blob/364dafc88412c3527cd49c9345404ccb64c68e20/tests/unit/amo/server/test_base.js#L557

Reply:
Maybe. I can give it a try. I guess one might ask why we're doing it for this and not for any other instances where we rely on `componentDidMount` not getting called on the server. I know I'm the one suggesting we should have a test for this, but I'm also anticipating that question from you. 😛